# Wendy

Import libiraries

In [1]:
library(dplyr)
library(lubridate)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



Attaching package: ‘lubridate’


The following objects are masked from ‘package:base’:

    date, intersect, setdiff, union




Import data

In [2]:
stock <- read.csv('compustat_food_bev.csv')
head(stock)

GVKEY  iid datadate   tic  conm                       cshtrd  prccd prchd
1 186785 1   01/09/2020 ARCO ARCOS DORADOS HOLDINGS INC  813895 4.47  4.540
2 186785 1   02/09/2020 ARCO ARCOS DORADOS HOLDINGS INC  518021 4.48  4.522
3 186785 1   03/09/2020 ARCO ARCOS DORADOS HOLDINGS INC  947825 4.44  4.525
4 186785 1   04/09/2020 ARCO ARCOS DORADOS HOLDINGS INC  534286 4.41  4.490
5 186785 1   08/09/2020 ARCO ARCOS DORADOS HOLDINGS INC  669380 4.27  4.415
6 186785 1   09/09/2020 ARCO ARCOS DORADOS HOLDINGS INC 1152416 4.38  4.420
  prcld prcod exchg sic 
1 4.394 4.41  11    5812
2 4.400 4.49  11    5812
3 4.360 4.48  11    5812
4 4.280 4.48  11    5812
5 4.260 4.34  11    5812
6 4.270 4.27  11    5812

In [3]:
distinct(stock, tic)

tic  
1  ARCO 
2  ARKR 
3  ARMK 
4  BBQ  
5  BDL  
6  BFI  
7  BH   
8  BH.A 
9  BJRI 
10 BLMN 
11 BROS 
12 BTBD 
13 CAKE 
14 CBRL 
15 CHUY 
16 CMG  
17 CNNE 
18 DENN 
19 DNKN 
20 DPZ  
21 DRI  
22 EAT  
23 FAT  
24 FATBB
25 FRGI 
26 FWRG 
27 GCFB 
28 GTIM 
29 JACK 
30 JAX  
31 KRUS 
32 LOCO 
33 LUB  
34 MCD  
35 NDLS 
36 PBPB 
37 PTLO 
38 PZZA 
39 QSR  
40 REBN 
41 RRGB 
42 RUTH 
43 SBUX 
44 SDOT 
45 SG   
46 SHAK 
47 STKS 
48 TACO 
49 TAST 
50 THCH 
51 TXRH 
52 WEN  
53 WING 
54 YOSH 
55 YUM  
56 YUMC

In [4]:
wendy = filter(stock, tic=="WEN")
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic 
1 5812
2 5812
3 5812
4 5812
5 5812
6 5812

Daily return

In [5]:
wendy=mutate(wendy, daily_return=(prccd-lag(prccd))/lag(prccd))
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic  daily_return 
1 5812            NA
2 5812  0.0486873508
3 5812 -0.0009103323
4 5812 -0.0168564920
5 5812  0.0125115848
6 5812  0.0295194508

Overnight return

In [6]:
wendy=mutate(wendy, overnight_return=(prcod-lag(prccd))/lag(prccd))
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic  daily_return  overnight_return
1 5812            NA            NA   
2 5812  0.0486873508  0.0028639618   
3 5812 -0.0009103323 -0.0009103323   
4 5812 -0.0168564920  0.0113895216   
5 5812  0.0125115848 -0.0069508804   
6 5812  0.0295194508  0.0068649886

Volume change

In [7]:
wendy=mutate(wendy, volume_change=(cshtrd-lag(cshtrd)))
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic  daily_return  overnight_return volume_change
1 5812            NA            NA         NA      
2 5812  0.0486873508  0.0028639618     884992      
3 5812 -0.0009103323 -0.0009103323     466079      
4 5812 -0.0168564920  0.0113895216    -929061      
5 5812  0.0125115848 -0.0069508804      87249      
6 5812  0.0295194508  0.0068649886    -591443

Money flow volume indicator (MFV)

In [8]:
wendy=mutate(wendy, mvf=(((prccd-prcld)-(prchd-prccd))/(prchd-prcld))*cshtrd)
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic  daily_return  overnight_return volume_change mvf     
1 5812            NA            NA         NA              0
2 5812  0.0486873508  0.0028639618     884992        3030530
3 5812 -0.0009103323 -0.0009103323     466079       -1153558
4 5812 -0.0168564920  0.0113895216    -929061       -2148667
5 5812  0.0125115848 -0.0069508804      87249        1761526
6 5812  0.0295194508  0.0068649886    -591443        2005638

New dataset

In [9]:
wendy_new=select(wendy, datadate, daily_return, overnight_return, volume_change, mvf)
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
1 3708  3   01/09/2020 WEN WENDY'S CO 2929911 20.950 21.16 20.7400 20.94 14   
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
  sic  daily_return  overnight_return volume_change mvf     
1 5812            NA            NA         NA              0
2 5812  0.0486873508  0.0028639618     884992        3030530
3 5812 -0.0009103323 -0.0009103323     466079       -1153558
4 5812 -0.0168564920  0.0113895216    -929061       -2148667
5 5812  0.0125115848 -0.0069508804      87249        1761526
6 5812  0.0295194508  0.0068649886    -591443        2005638

Drop first row because it have nan values

In [10]:
wendy = wendy[2:nrow(wendy),]
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
2 3708  3   02/09/2020 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   03/09/2020 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   04/09/2020 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   08/09/2020 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   09/09/2020 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
7 3708  3   10/09/2020 WEN WENDY'S CO 2787857 21.850 23.04 21.8400 22.55 14   
  sic  daily_return  overnight_return volume_change mvf     
2 5812  0.0486873508  0.0028639618     884992        3030530
3 5812 -0.0009103323 -0.0009103323     466079       -1153558
4 5812 -0.0168564920  0.0113895216    -929061       -2148667
5 5812  0.0125115848 -0.0069508804      87249        1761526
6 5812  0.0295194508  0.0068649886    -591443        2005638
7 5812 -0.0286730385  0.0024449878     -59870       -2741393

Month column

In [11]:
wendy$datadate = as.Date(wendy$datadate, format="%d/%m/%Y")
wendy = mutate(wendy, month=month(wendy$datadate))
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
2 3708  3   2020-09-02 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   2020-09-03 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   2020-09-04 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   2020-09-08 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   2020-09-09 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
7 3708  3   2020-09-10 WEN WENDY'S CO 2787857 21.850 23.04 21.8400 22.55 14   
  sic  daily_return  overnight_return volume_change mvf      month
2 5812  0.0486873508  0.0028639618     884992        3030530 9    
3 5812 -0.0009103323 -0.0009103323     466079       -1153558 9    
4 5812 -0.0168564920  0.0113895216    -929061       -2148667 9    
5 5812  0.0125115848 -0.0069508804      87249        1761526 9    
6 5812  0.0295194508  0.0068649886    -591443        2005638 9    
7 5812 -0.0286730385  0.0024449878     -59870       -2741393 9

Year column

In [12]:
wendy = mutate(wendy, year=year(wendy$datadate))
head(wendy)

GVKEY iid datadate   tic conm       cshtrd  prccd  prchd prcld   prcod exchg
2 3708  3   2020-09-02 WEN WENDY'S CO 3814903 21.970 22.08 21.0100 21.01 14   
3 3708  3   2020-09-03 WEN WENDY'S CO 4280982 21.950 22.48 21.6450 21.95 14   
4 3708  3   2020-09-04 WEN WENDY'S CO 3351921 21.580 22.38 21.4050 22.20 14   
5 3708  3   2020-09-08 WEN WENDY'S CO 3439170 21.850 22.05 21.2300 21.43 14   
6 3708  3   2020-09-09 WEN WENDY'S CO 2847727 22.495 22.61 21.8322 22.00 14   
7 3708  3   2020-09-10 WEN WENDY'S CO 2787857 21.850 23.04 21.8400 22.55 14   
  sic  daily_return  overnight_return volume_change mvf      month year
2 5812  0.0486873508  0.0028639618     884992        3030530 9     2020
3 5812 -0.0009103323 -0.0009103323     466079       -1153558 9     2020
4 5812 -0.0168564920  0.0113895216    -929061       -2148667 9     2020
5 5812  0.0125115848 -0.0069508804      87249        1761526 9     2020
6 5812  0.0295194508  0.0068649886    -591443        2005638 9     2020
7 5812 -0.0286730385  0.0024449878     -59870       -2741393 9     2020

Total trading volume in june 2023

In [13]:
wendy_jun_2023=filter(wendy, month==6 & year==2023)
sum(wendy_jun_2023$cshtrd)
paste0("The total trading volume in june 2023 was ", sum(wendy_jun_2023$cshtrd))

[1] 54557454

[1] "The total trading volume in june 2023 was 54557454"

Mean daily return

In [14]:
mean(wendy$daily_return)
paste0("The mean daily return, over the entire period was ", mean(wendy$daily_return))

[1] 0.0001164747

[1] "The mean daily return, over the entire period was 0.000116474712706267"

The date that saw the largest positive high price.

In [15]:
max(wendy$prchd)
filter(wendy, prchd==max(wendy$prchd))
paste0("The date that saw the largest positive high price was ", filter(wendy, daily_return==max(wendy$daily_return))$datadate)

[1] 29.46

GVKEY iid datadate   tic conm       cshtrd   prccd prchd prcld prcod  exchg
1 3708  3   2021-06-08 WEN WENDY'S CO 86616410 28.87 29.46 25.39 27.345 14   
  sic  daily_return overnight_return volume_change mvf      month year
1 5812 0.2585004    0.1920227        83374146      61504036 6     2021

[1] "The date that saw the largest positive high price was 2021-06-08"

The date that saw the largest positive daily return.

In [16]:
max(wendy$daily_return)
filter(wendy, daily_return==max(wendy$daily_return))
print(paste0("The date that saw the largest positive daily return was ", filter(wendy, daily_return==max(wendy$daily_return))$datadate))

[1] 0.2585004

GVKEY iid datadate   tic conm       cshtrd   prccd prchd prcld prcod  exchg
1 3708  3   2021-06-08 WEN WENDY'S CO 86616410 28.87 29.46 25.39 27.345 14   
  sic  daily_return overnight_return volume_change mvf      month year
1 5812 0.2585004    0.1920227        83374146      61504036 6     2021

[1] "The date that saw the largest positive daily return was 2021-06-08"


"Shares of The Wendy's Company (NASDAQ: WEN) were rocketing 18% higher today after the WallStreetBets subreddit crowd apparently turned its attention to the fast-food chain and mentioned it as a possible short squeeze candidate" (Duprey, 2021).

Duprey, R. (2021) Wendy's is Suddenly a Meme Stock. Available at: https://www.nasdaq.com/articles/wendys-is-suddenly-a-meme-stock-2021-06-08 (Accessed: 25 September 2025).
